hal/armv7m/pmap.c

```diff
@@ -36,9 +49,53 @@ addr_t pmap_destroy(pmap_t *pmap, int *i)
 }
 
 
+static int pmap_map2region(unsigned int map)
+{
+	int i;
+
+	for (i = 0; i < sizeof(syspage->hs.mpu.map) / sizeof(*syspage->hs.mpu.map); ++i) {
+		if (map == syspage->hs.mpu.map[i]) {
+			return i;
+		}
+	}
+
+	return -1;
+}
+
+
+int pmap_addMap(pmap_t *pmap, unsigned int map)
+{
+	int region = pmap_map2region(map);
+	if (region < 0) {
+		return region;
+	}
+
+	pmap->regions |= (1 << region);
+
+	return 0;
+}
+
+
 void pmap_switch(pmap_t *pmap)
 {
-	return;
+	unsigned int i;
+
+	if (pmap != NULL) {
+		for (i = 0; i < syspage->hs.mpu.allocCnt; ++i) {
```

Comment:
What do you think about this solution, faster and smaller result code.
```suggestion
	unsigned int i;
	const unsigned int allocCnt = syspage->hs.mpu.allocCnt;

	if (pmap != NULL) {
		for (i = 0; i < allocCnt; ++i) {
```

Reply:
Thanks, will use, for now I still need to debug some problems.
Btw, nice to see you back ;)